### PR TITLE
Rework intro, prerequs and execution sections for flow and installation options

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description

Some changes to the overview and prerequisites pages to make the flow and content clearer, and to take steps to account for local/cloud/air-gapped installs.

Further work is planned to make seaparate guides for these installation types.

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/server/v4.4/installation/phase-1-prerequisites.adoc [hunk 1]
```diff
@@ -49,40 +44,71 @@ Download and install the following software before continuing:
 | {helmdiffversion} or greater
 | Helping with `values.yaml` changes and updates
 | Optional, but may help with troubleshooting between releases
+
+| link:https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html[AWS CLI]
```
Comment (rosieyohannan) [inline jekyll/_cci2/server/v4.4/installation/phase-1-prerequisites.adoc:48]: Need some help to add versions and uses here - maybe versions aren't important as they weren't documented previously

File: jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc [hunk 2]
```diff
@@ -19,11 +19,17 @@ CircleCI server is an on-premises CI/CD platform for enterprise customers who ha
...
+
+The following diagram presents and overview of the services architecture of CircleCI server.
 
 .CircleCI server v4.4 Architecture
 image::server-4-architecture-diagram.png[Server v4.4 services architecture]
 
+[#ports-and-services]
+=== Ports and services
+
 The CircleCI server application exposes four services, using a single load balancer. If required, the load balancer can be made private, separating it from the public internet.
```
Comment (rosieyohannan) [inline jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc:33]: Is this still correct? Four?

File: jekyll/_cci2/server/v4.4/installation/phase-1-prerequisites.adoc (hunk 1 above)
Comment (ryanwohara) [reply jekyll/_cci2/server/v4.4/installation/phase-1-prerequisites.adoc:48]: Latest should be fine

File: jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc (hunk 2 above)
Comment (ryanwohara) [reply jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc:33]: Ah good catch. This is now just two services:
* nginx
* nomad-server

File: jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc [hunk 3]
```diff
@@ -361,6 +372,15 @@ We do our best to support a wide range of platforms for installation. We use env
 | Should work with MinIO and Runner.
 |===
 
+[#installation-options]
+== Installation options
+
+CircleCI server can be installed using cloud resrouces (GCP or AWS), locally, and in an air-gapped environment. For installation instructions, refer to the following:
```
Comment (ryanwohara) [inline jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc:378]:
```suggestion
CircleCI server can be installed using cloud resources (GCP or AWS), locally, and in an air-gapped environment. For installation instructions, refer to the following:
```

File: jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc [hunk 4]
```diff
@@ -83,13 +89,18 @@ The application exposes a number of external ports. These ports are used for var
 | Only required for the Nomad clients.
 |===
 
-CircleCI server schedules CI jobs using the link:https://www.nomadproject.io/[Nomad] scheduler. The Nomad control plane runs inside of Kubernetes, while the Nomad clients, which are responsible for running scheduled CircleCI jobs, are provisioned outside the cluster. CircleCI
-server can run Docker jobs on the Nomad clients themselves or in a dedicated virtual machine (VM).
+[#job-scheduling-and-artifact-storage]
+=== Job scheduling and artifact storage
+
+CircleCI server schedules CI jobs using HashiCorp link:https://www.nomadproject.io/[Nomad]. The Nomad control plane runs inside your Kubernetes cluster, while the Nomad clients, which are responsible for running scheduled CircleCI jobs, are provisioned outside the cluster.
```
Comment (ryanwohara) [inline jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc:95]: This is less true now. We do still use Nomad, but not like we used to. In the past the VM jobs were orchestrated through Nomad but that is no longer the case.
Now we have three different systems that jobs can utilize:
* Nomad
* VMs (via Machine Provisioner)
* Runner

File: jekyll/_cci2/server/v4.4/installation/phase-1-prerequisites.adoc (hunk 1 above)
Comment (rosieyohannan) [reply jekyll/_cci2/server/v4.4/installation/phase-1-prerequisites.adoc:48]: 👍

File: jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc (hunk 2 above)
Comment (rosieyohannan) [reply jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc:33]: 👍

File: jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc (hunk 4 above)
Comment (rosieyohannan) [reply jekyll/_cci2/server/v4.4/overview/circleci-server-overview.adoc:95]: @ryanwohara so Nomad is just for Docker executor?

File: jekyll/_cci2/server/v4.4/installation/phase-3-execution-environments.adoc [hunk 5]
```diff
@@ -17,12 +17,26 @@ Before you begin with the CircleCI server v4.4 execution environment installatio
 
 NOTE: In the following sections, replace any sections indicated by `< >` with your details.
 
+== Introduction
+
+Execution environments in CircleCI server are resourced in the following ways:
+
+* HashiCorp Nomad is used for scheduling jobs and running Docker jobs
+** Nomad Server is installed and configued within your Kubernetes cluster and is used for scheduling jobs
```
Comment (ryanwohara) [inline jekyll/_cci2/server/v4.4/installation/phase-3-execution-environments.adoc:25]:
```suggestion
** Nomad Server is installed and configured within your Kubernetes cluster and is used for scheduling jobs
```

File: jekyll/_cci2/server/v4.4/installation/phase-3-execution-environments.adoc [hunk 6]
```diff
@@ -17,12 +17,26 @@ Before you begin with the CircleCI server v4.4 execution environment installatio
...
+** Nomad Clients are installed outside your Kubernetes cluster and are used for running jobs that are configured to use the Docker execution environment. CircleCI provides sample Terraform modules for you to reference and use to streamline the installation process if you are installing within AWS or GCP. Installing in this way also enables the use of the Nomad Autoscaler
+
+* <<machine-provisioner,Machine provisioner>> is used in AWS/GCP installation to provision resources and run jobs that are configured to use the `machine` execution environment, including Linux and Windows VMs. If you install outside AWS or GCP (that is, a local installation) you will use self-hosted runners to run VM-configured jobs.
```
Comment (ryanwohara) [inline jekyll/_cci2/server/v4.4/installation/phase-3-execution-environments.adoc:28]:
```suggestion
* <<machine-provisioner,Machine provisioner>> is used in AWS/GCP installations to provision resources and run jobs that are configured to use the `machine` execution environment, including Linux and Windows VMs. If you install outside AWS or GCP (that is, a local installation) you will use self-hosted runners to run VM-configured jobs.
```

I kind of want to mention ARM support here, though it's AWS only

**Pull request merged**